src/sentry/incidents/endpoints/organization_alert_rule_available_action_index.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import absolute_import
+
+from collections import defaultdict
+
+from rest_framework import status
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.incidents.endpoints.bases import OrganizationEndpoint
+from sentry.incidents.endpoints.serializers import action_target_type_to_string
+from sentry.incidents.logic import get_available_action_integrations_for_org
+from sentry.incidents.models import AlertRuleTriggerAction
+
+
+class OrganizationAlertRuleAvailableActionIndexEndpoint(OrganizationEndpoint):
+    def build_action_response(self, registered_type, integration=None):
+        return {
+            "type": registered_type.slug,
+            "allowedTargetTypes": [
+                action_target_type_to_string[target_type]
+                for target_type in registered_type.supported_target_types
```

Comment:
Should the registered_type objects expose their target type names? Having multiple modules coupling to a mutable module level dictionary feels dangerous.

Reply:
Maybe I should just convert these to frozensets so that we don't have to worry about them being mutable

Reply:
I did this in the PR that implements this registry, which has been merged to master.